velox/functions/prestosql/window/tests/WindowTestBase.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/functions/prestosql/window/WindowFunctionsRegistration.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::window::test {
+
+class WindowTestBase : public OperatorTestBase {
+ protected:
+  void SetUp() override {
+    exec::test::OperatorTestBase::SetUp();
+    velox::window::registerWindowFunctions();
+  }
+
+  std::vector<RowVectorPtr>
+  makeVectors(const RowTypePtr& rowType, vector_size_t size, int numVectors) {
+    std::vector<RowVectorPtr> vectors;
+    VectorFuzzer::Options options;
+    options.vectorSize = size;
+    VectorFuzzer fuzzer(options, pool_.get(), 0);
+    for (int32_t i = 0; i < numVectors; ++i) {
+      auto vector =
+          std::dynamic_pointer_cast<RowVector>(fuzzer.fuzzRow(rowType));
+      vectors.push_back(vector);
+    }
+    return vectors;
+  }
+
+  void testWindowFunction(
+      const std::vector<RowVectorPtr>& input,
+      const std::string& function,
+      const std::string& overClause) {
+    auto functionSql = fmt::format("{}() over ({})", function, overClause);
+
+    SCOPED_TRACE(functionSql);
+    auto op = PlanBuilder().values(input).window({functionSql}).planNode();
+
+    auto rowType = asRowType(input[0]->type());
+    std::string columnsString = folly::join(", ", rowType->names());
+    columnsString += ", ";
+
+    assertQuery(
+        op, fmt::format("SELECT {} {} FROM tmp", columnsString, functionSql));
+  };
+
+  void testWindowFunction(
+      const std::vector<RowVectorPtr>& input,
+      const std::string& function,
+      const std::vector<std::string> overClauses) {
+    for (const auto& overClause : overClauses) {
+      testWindowFunction(input, function, overClause);
+    }
+  }
+
+  // This function operates on input RowVectors that have 2 columns.
+  // It verifies (for the windowFunction) SQL queries with varying over
+  // clauses. The over clauses covers all combinations of partition by
+  // and order by of the two input columns.
+  void testTwoColumnInput(
```

Comment:
Can this function generate the necessary inputs ? Do these need to be provided by the caller?

Reply:
So far I have varied the input tables to make the input interesting for the function.

When we settle on a generic function we would also generalize the input data construction.